pkg/providers/cloudstack/validator.go

```diff
@@ -190,6 +194,22 @@ func (v *Validator) validateMachineConfig(ctx context.Context, datacenterConfig
 	return nil
 }
 
+func (v *Validator) validateTemplateMatchesKubernetesVersion(ctx context.Context, machineConfig *anywherev1.CloudStackMachineConfig, spec *cluster.Spec) error {
+	// Replace 1.23, 1-23 to 123 in the template name string.
+	templateReplacer := strings.NewReplacer("-", "", ".", "", "_", "")
+	templateName := templateReplacer.Replace(machineConfig.Spec.Template.Name)
+	// Replace 1-23 to 123 in the kubernetesversion string.
+	replacer := strings.NewReplacer(".", "")
+	kubernetesVersion := replacer.Replace(string(spec.Cluster.Spec.KubernetesVersion))
+	// This will return an error if the template name does not contain specified kubernetes version.
+	// For ex if the kubernetes version is 1.23,
+	// the template name should include 1.23 or 1-23 or 123 i.e. kubernetes-1-23-eks in the string.
+	if !strings.Contains(templateName, kubernetesVersion) {
+		return fmt.Errorf("invalid template: cluster kubernetes version is %s but template for machineconfig %s is %s", string(spec.Cluster.Spec.KubernetesVersion), machineConfig.Name, machineConfig.Spec.Template.Name)
```

Comment:
same comment as https://github.com/aws/eks-anywhere/pull/6728#discussion_r1336417088 here
could we add this info to the error message?
the comment above is great!